### PR TITLE
Syntax errors for invalid 128-bit integer literals

### LINKED ISSUE #10962: Crash when compiling negative u128 constants
Hello lovely people of the internet!

Cutting to the point, I've managed to outsmart the compiler.
Tested on dev Crystal aswell, compiling the proceding code (simply `echo -1_u128|crystal eval`) will immediately crash the compiler.

```crystal
puts -1_u128
```

I'm really unfamilliar with the compiler's infrastructure, I can't help but give my unfounded guess about what's happening, and that I won't do, sorry! What I _can_ say, is that

**The code `puts -1_u64` seems to go through the compiler just fine. ( it does error out, as expected. _( it's expected behaviour, right? )_ )**

What follows is the output of running `echo -1_u128 | crystal eval` (from Release Crystal), but I presume the output will be replicable easily anyhow
```
Invalid UInt64: -1 (ArgumentError)
  from /crystal/src/string.cr:477:5 in 'to_u64'
  from /crystal/src/compiler/crystal/codegen/codegen.cr:458:24 in 'visit'
  from /crystal/src/compiler/crystal/syntax/visitor.cr:27:12 in 'accept'
  from /crystal/src/compiler/crystal/codegen/codegen.cr:644:9 in 'accept'
  from /crystal/src/compiler/crystal/codegen/codegen.cr:2219:7 in 'codegen'
  from /crystal/src/compiler/crystal/compiler.cr:172:16 in 'compile'
  from /crystal/src/compiler/crystal/command/eval.cr:30:5 in 'eval'
  from /crystal/src/compiler/crystal/command.cr:92:7 in 'run'
  from /crystal/src/compiler/crystal.cr:11:1 in '__crystal_main'
  from /crystal/src/crystal/main.cr:110:5 in 'main'
  from src/env/__libc_start_main.c:94:2 in 'libc_start_main_stage2'
Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues
```

I'm loving every bit about Crystal, Keep the amazing work up ❤️

### PR DESCRIPTION
**Pull request by rymiel**
Fixes #10962 

Since I am not skilled enough to resolve these TODOs in the codegen, I simply opted to warn about not being allowed to use literals larger than a 64-bit integer
https://github.com/crystal-lang/crystal/blob/71d6e9b63fa06b3ace5207fc0ea1f1d20a2c6089/src/compiler/crystal/codegen/codegen.cr#L453-L458

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]

**Comment by straight-shoota** [comment]
Thanks. Parser warnings are really fine for now. Full 128-bit support is a more complex story (#8373).

File: src/compiler/crystal/syntax/lexer.cr [hunk 1]
```diff
@@ -1575,17 +1575,25 @@ module Crystal
...
     end
 
-    macro gen_check_int_fits_in_size(type, method, size)
+    macro gen_check_int_fits_in_size(type, method, size, *, actual_type = nil)
       if num_size >= 20
+        {% if actual_type.nil? %}
         raise_value_doesnt_fit_in "{{type}}", string_value, start
+        {% else %}
+        raise_value_restricted_by "{{actual_type}}", "{{type}}", string_value, start
+        {% end %}
```
Comment (Sija) [inline src/compiler/crystal/syntax/lexer.cr:1584]:
```suggestion
        {% if actual_type.nil? %}
          raise_value_doesnt_fit_in "{{type}}", string_value, start
        {% else %}
          raise_value_restricted_by "{{actual_type}}", "{{type}}", string_value, start
        {% end %}
```

File: src/compiler/crystal/syntax/lexer.cr [hunk 2]
```diff
@@ -1575,17 +1575,25 @@ module Crystal
...
         int_value = absolute_integer_value(string_value, negative)
         max = {{type}}::MAX.{{method}}
         max += 1 if negative
 
         if int_value > max
+          {% if actual_type.nil? %}
           raise_value_doesnt_fit_in "{{type}}", string_value, start
+          {% else %}
+          raise_value_restricted_by "{{actual_type}}", "{{type}}", string_value, start
+          {% end %}
```
Comment (Sija) [inline src/compiler/crystal/syntax/lexer.cr:1596]:
```suggestion
          {% if actual_type.nil? %}
            raise_value_doesnt_fit_in "{{type}}", string_value, start
          {% else %}
            raise_value_restricted_by "{{actual_type}}", "{{type}}", string_value, start
          {% end %}
```

**Review by Sija — COMMENTED** [review]
Missing indentation suggestions.

**Review by straight-shoota — APPROVED** [review]

**Review by sdogruyol — APPROVED** [review]
Thank you @rymiel 🙏

**Pull request merged**